### PR TITLE
+ Add configuration for The Graph

### PR DESCRIPTION
**Pull request by factotumn**
Streamline onboarding.

### CODE REVIEW

**Comment by peripheralist** [comment]
thanks for checking this! Though let's not put a specific URL in the readme. In some cases we'll want to use different URLs (like with testing on Rinkeby, or if using your own API key).

Maybe something like:
> To integrate with The Graph, set `REACT_APP_SUBGRAPH_URL` in your `.env` file to a query URL for the Juicebox subgraph you'd like to use.

**Comment by cunningproof** [comment]
Totally. That URL was all I found after searching the discord - is there a definitive list we can point people to?

File: README.md [hunk 1]
```diff
@@ -85,7 +85,10 @@ Take the following steps to set up Blocknative for local development:
    - Enable the **Admin** toggle in the **Admin** field.
 1. Copy the API key into the `REACT_APP_BLOCKNATIVE_API_KEY` variable of the
    `.env` file.
+#### The Graph
+JuiceBox uses [The Graph](https://thegraph.com/) as an additional data provider.
 
+To integrate with The Graph, set the `REACT_APP_SUBGRAPH_URL` in your `.env` to a query URL for the Juicebox subgraph you'd like to use.
```
Comment (aeolianeth) [inline README.md:91]: **suggestion**: perhaps we could give some instruction about how to find the URL:

```suggestion
To set up The Graph locally, first determine the subgraph you'd like to use. Then, set the `REACT_APP_SUBGRAPH_URL` in your `.env` to the subgraph URL.

The subgraph you use depends on which Ethereum network you're using. Juicebox has subgraphs for `mainnet` and `rinkeby`. Subgraphs are also versioned. Typically, you'll want to use the latest version of the subgraph.

For `mainnet`:
1. Find and select Juicebox on [The Graph Explorer](https://thegraph.com/explorer).
1. Find the Deployment ID of the subgraph.
1. Set `REACT_APP_SUBGRAPH_URL` to `https://api.thegraph.com/subgraphs/id/<Deployment ID>` in your `.env` file.

For `rinkeby`, ask one the friendly Juicebox devs in [Discord](https://discord.gg/6jXrJSyDFf).
```

cc @peripheralist

**Review by aeolianeth — CHANGES_REQUESTED** [review]
@cunningproof hope you don't mind me leaving a suggestion! Let me know what you think 🙏

File: README.md (hunk 1 above)
Comment (cunningproof) [reply README.md:91]: Not at all! Excellent improvement.

**Comment by aeolianeth** [comment]
Thanks @cunningproof! LGTM, handing to @peripheralist for final review 🏓

**Review by aeolianeth — APPROVED** [review]

**Comment by aeolianeth** [comment]
FWIW, it's still unclear to me about which URL we should actually use.

My understanding is that The Graph actually suggests we use this scheme with an API key:

`https://gateway.thegraph.com/api/[api-key]/subgraphs/id/0x63a2368f4b509438ca90186cb1c15156713d5834-0`

But, to create an API key, you need to pay (with GRT).

The scheme suggested in this PR is "free" to use: https://github.com/jbx-protocol/juice-interface/pull/268/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R98. But I can't find anywhere in The Graph docs that suggest this approach 😅

**Comment by peripheralist** [comment]
How about we just point folks to the subgraph page itself (https://thegraph.com/explorer/subgraph?id=0x63a2368f4b509438ca90186cb1c15156713d5834-0) and also include a reference to the repo (https://github.com/jbx-protocol/juice-subgraph)?

Eventually we'll want the official JB API key to be restricted to the juicebox.money domain, meaning it won't work in dev. Once that happens we'll just keep an internal API key with a dao-funded balance for JB devs to use.

For rinkeby, there isn't an official subgraph since publishing testnet subgraphs isn't reliable. For the same reason, we only ever use the dev query subgraph URL which doesn't require an API key. Theoretically anyone could fork the JB subgraph from the repo and publish their own—not sure that would ever make sense for mainnet, but it might for testnet, especially when working on features that might require subgraph changes.

I'm open to including a link to the rinkeby subgraph query url, but since we'd also need to remember to keep it updated I think I'd prefer to just request that folks ask the dev team in the discord.

**Comment by peripheralist** [comment]
hey @factotumn did you want to stay on this or should we close it? not a very critical thing, so no worries if you're distracted elsewhere now

**Comment by factotumn** [comment]
> hey @factotumn did you want to stay on this or should we close it? not a very critical thing, so no worries if you're distracted elsewhere now

I will follow through on it

**Comment by factotumn** [comment]
> The scheme suggested in this PR is "free" to use: https://github.com/jbx-protocol/juice-interface/pull/268/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R98. But I can't find anywhere in The Graph docs that suggest this approach 😅

Yeah; I found this URL by grepping the discord

**Comment by factotumn** [comment]
> hey @factotumn did you want to stay on this or should we close it? not a very critical thing, so no worries if you're distracted elsewhere now

I will review your latest suggestion for changes and finish this one.

**Comment by 0xzachvai** [comment]
I'm trying to onboard to the project as a new contributor, and so far I've been able to get a key for everything but thegraph which, like @peripheralist mentioned, doesn't seem to be free.

What's the recommended approach for now?

**Comment by peripheralist** [comment]
@0xzachvai for now you can use the URL from the production juicebox app, but this pattern will be changing soon.

hop in the juicebox discord https://discord.gg/juicebox if you want a hand getting set up!

**Pull request closed without merging**